pkg/oci/repository.go

```diff
@@ -51,16 +48,119 @@ type ResourceOptions struct {
 	SnapshotName     string
 }
 
+// ClientOptsFunc options are used to leave the cache backwards compatible.
+// If the certificate isn't defined, we will use `WithInsecure`.
+type ClientOptsFunc func(opts *Client)
+
+// WithCertificateSecret defines the name of the secret holding the certificates.
+func WithCertificateSecret(name string) ClientOptsFunc {
+	return func(opts *Client) {
+		opts.CertSecretName = name
+	}
+}
+
+// WithNamespace sets up certificates for the client.
+func WithNamespace(namespace string) ClientOptsFunc {
+	return func(opts *Client) {
+		opts.Namespace = namespace
+	}
+}
+
+// WithInsecureSkipVerify sets up certificates for the client.
+func WithInsecureSkipVerify(value bool) ClientOptsFunc {
+	return func(opts *Client) {
+		opts.InsecureSkipVerify = value
+	}
+}
+
+// WithClient sets up certificates for the client.
+func WithClient(client client.Client) ClientOptsFunc {
+	return func(opts *Client) {
+		opts.Client = client
+	}
+}
+
 // Client implements the caching layer and the OCI layer.
 type Client struct {
-	OCIRepositoryAddr string
+	Client             client.Client
+	OCIRepositoryAddr  string
+	InsecureSkipVerify bool
+	Namespace          string
+	CertSecretName     string
+
+	certPem []byte
+	keyPem  []byte
+	ca      []byte
+}
+
+// WithTransport sets up insecure TLS so the library is forced to use HTTPS.
+func (c *Client) WithTransport() Option {
```

Comment:
we should split this function, it does too many things

Reply:
👍